builder/builder.go

```diff
@@ -850,3 +874,70 @@ func parseUint32(s string) (uint32, error) {
 	}
 	return uint32(parsed), nil
 }
+
+// createCompressedArchive will use tar and xz to create a compressed
+// file. It does not stream the sources contents, doing all the work
+// in memory before writing the destination file.
+func createCompressedArchive(dst string, srcs ...string) error {
+	err := createCompressedArchiveInternal(dst, srcs...)
+	return errors.Wrapf(err, "couldn't create compressed archive %s", dst)
+}
+
+func createCompressedArchiveInternal(dst string, srcs ...string) error {
+	archive := &bytes.Buffer{}
+	xw, err := swupd.NewExternalWriter(archive, "xz")
+	if err != nil {
+		return err
+	}
+
+	err = archiveFiles(xw, srcs)
+	if err != nil {
+		_ = xw.Close()
+		return err
+	}
+
+	err = xw.Close()
+	if err != nil {
+		return err
+	}
+
+	err = ioutil.WriteFile(dst, archive.Bytes(), 0644)
+	if err != nil {
```

Comment:
just return err here and remove last 5 lines?

Reply:
Fixed.